### PR TITLE
docs: add proxy observability doc

### PR DESCRIPTION
**Pull request by zirain**

### CODE REVIEW

File: docs/latest/user/proxy-observability.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+By default, Envoy Gateway doesn't expose metrics of the EnvoyProxy instances. 
+You can enable metrics by setting the `telemetry.metrics.prometheus` in the `EnvoyProxy` CRD.
+
+Expose prometheus metrics endpoints:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/envoyproxy/gateway/latest/examples/kubernetes/metric/prometheus.yaml
+```
+
+Verify metrics:
```
Comment (arkodg) [inline docs/latest/user/proxy-observability.md:60]: can we add grafana and a stock envoy dashboard in here ?

Comment (zirain) [reply docs/latest/user/proxy-observability.md:60]: need more work and screenshot, it can be separated.

Comment (arkodg) [reply docs/latest/user/proxy-observability.md:60]: cool lets track it with a GH issue

**Comment by arkodg** [comment]
hey @zirain can we also copy this content over to `docs/v0.5.0` to it shows up in our current docs ?

File: docs/latest/user/proxy-observability.md (hunk 1 above)
Comment (arkodg) [reply docs/latest/user/proxy-observability.md:60]: raised https://github.com/envoyproxy/gateway/issues/1764 to track this

**Comment by zirain** [comment]
> hey @zirain can we also copy this content over to `docs/v0.5.0` to it shows up in our current docs ?

done

**Review by arkodg — DISMISSED** [review]
thanks !

File: docs/latest/user/proxy-observability.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
+# Proxy Observability
+
+Envoy Gateway provide observability for the ControlPlane and the underlying EnvoyProxy instances.
```
Comment (Xunzhuo) [inline docs/latest/user/proxy-observability.md:3]: provides

File: docs/v0.5.0/user/proxy-observability.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
+# Proxy Observability
+
+Envoy Gateway provide observability for the ControlPlane and the underlying EnvoyProxy instances.
```
Comment (Xunzhuo) [inline docs/v0.5.0/user/proxy-observability.md:3]: ditto

File: docs/latest/user/proxy-observability.md (hunk 2 above)
Comment (zirain) [reply docs/latest/user/proxy-observability.md]: done

**Review by Xunzhuo — APPROVED** [review]

**Review by arkodg — APPROVED** [review]

**Pull request merged**